Tasks/BeanstalkDeployApplication/helpers/DeployApplicationTaskOperations.ts

```diff
@@ -163,36 +168,54 @@ export class TaskOperations {
 
         let success = true;
         let environment: Beanstalk.EnvironmentDescription;
+
+        // delay the event poll by a random amount, up to 5 seconds, so that if multiple
+        // deployments run in parallel they don't all start querying at the same time and
+        // potentially trigger throttling from the outset
+        const initialStartDelay = Math.floor(Math.random() * (randomJitterUpperLimit)) + 1;
```

Comment:
Consider adding an issue to the backlog to use a throttler here instead of a random delay.

Reply:
Filed https://github.com/aws/aws-vsts-tools/issues/171